src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.tsx

```diff
@@ -0,0 +1,49 @@
+import { shallow } from "enzyme";
+import * as React from "react";
+
+import { Carousel, CarouselProps } from "../Carousel";
+
+describe("Carousel component", () => {
+    const staticImages = [
+        { imageUrl: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" },
+        { imageUrl: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" }
+    ];
+    const render = (props: CarouselProps) => shallow(<Carousel {...props} />);
+
+    it("should render carousel structure", () => {
+        const output = render({ images: staticImages });
+        expect(output).toMatchStructure(
+            <div className="carousel">
+                <div className="carousel-inner" />
+            </div>
+        );
+    });
+
+    it("should render carousel with 2 children", () => {
+        const wrapper = render({ images: staticImages });
+        expect(wrapper.find(".carousel-inner").children().length).toEqual(staticImages.length);
```

Comment:
This is not a very useful test as you don't verify that the correct carousel images are made.